x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/DuplicateLimitAfterMvExpand.java

```diff
@@ -1,108 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
- */
-
-package org.elasticsearch.xpack.esql.optimizer.rules.logical;
-
-import org.elasticsearch.xpack.esql.core.expression.AttributeSet;
-import org.elasticsearch.xpack.esql.core.expression.ReferenceAttribute;
-import org.elasticsearch.xpack.esql.plan.logical.Aggregate;
-import org.elasticsearch.xpack.esql.plan.logical.Enrich;
-import org.elasticsearch.xpack.esql.plan.logical.Eval;
-import org.elasticsearch.xpack.esql.plan.logical.Filter;
-import org.elasticsearch.xpack.esql.plan.logical.Limit;
-import org.elasticsearch.xpack.esql.plan.logical.LogicalPlan;
-import org.elasticsearch.xpack.esql.plan.logical.MvExpand;
-import org.elasticsearch.xpack.esql.plan.logical.OrderBy;
-import org.elasticsearch.xpack.esql.plan.logical.Project;
-import org.elasticsearch.xpack.esql.plan.logical.RegexExtract;
-import org.elasticsearch.xpack.esql.plan.logical.UnaryPlan;
-
-public final class DuplicateLimitAfterMvExpand extends OptimizerRules.OptimizerRule<Limit> {
-
-    @Override
-    protected LogicalPlan rule(Limit limit) {
-        var child = limit.child();
-        var shouldSkip = child instanceof Eval
-            || child instanceof Project
-            || child instanceof RegexExtract
-            || child instanceof Enrich
-            || child instanceof Limit;
-
-        if (shouldSkip == false && child instanceof UnaryPlan unary) {
-            MvExpand mvExpand = descendantMvExpand(unary);
```

Comment:
This rule seems to cover more cases than simply LIMIT->MvExpand, which the other rule now covers. Presumably the iterative application of the other rules will cover this scenario already, so this removal is of no consequence. And presumably test coverage will pick this up if that is not the case.